pkg/server/init_deployer_runtime.go

```diff
@@ -0,0 +1,205 @@
+		return err
+	}
+
+	return nil
+}
+
+// applyDeployPermission applies the Kubernetes RBAC resources for deployer running.
+func applyDeployPermission(ctx context.Context, cli *kubernetes.Clientset) error {
+	sa := core.ServiceAccount{
+		ObjectMeta: meta.ObjectMeta{
+			Namespace: types.WalrusSystemNamespace,
+			Name:      types.DeployerServiceAccountName,
+		},
+	}
+
+	_, err := cli.CoreV1().
+		ServiceAccounts(sa.Namespace).
+		Create(ctx, &sa, meta.CreateOptions{})
+	if err != nil && !kerrors.IsAlreadyExists(err) {
+		return err
+	}
+
+	r := rbac.Role{
+		ObjectMeta: meta.ObjectMeta{
+			Namespace: types.WalrusSystemNamespace,
+			Name:      types.DeployerServiceAccountName,
+		},
+		Rules: []rbac.PolicyRule{
+			{
+				APIGroups: []string{batch.GroupName},
+				Resources: []string{"jobs"},
+				Verbs:     []string{rbac.VerbAll},
+			},
+			{
+				APIGroups: []string{core.GroupName},
+				Resources: []string{"secrets", "pods", "pods/log", "events"},
+				Verbs:     []string{rbac.VerbAll},
+			},
+			{
+				APIGroups: []string{coordination.GroupName},
+				Resources: []string{"leases"},
+				Verbs:     []string{rbac.VerbAll},
+			},
+		},
+	}
+
+	r_, err := cli.RbacV1().
+		Roles(r.Namespace).
+		Get(ctx, r.Name, meta.GetOptions{
+			ResourceVersion: "0",
+		})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return err
+	}
+
+	switch {
+	case r_ == nil || r_.Name == "" || r_.DeletionTimestamp != nil:
+		// Create.
+		_, err = cli.RbacV1().
+			Roles(r.Namespace).
+			Create(ctx, &r, meta.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	case !reflect.DeepEqual(r.Rules, r_.Rules):
+		// Update.
+		r.Labels = r_.Labels
+		r.Annotations = r_.Annotations
+		r.ResourceVersion = r_.ResourceVersion
+
+		_, err = cli.RbacV1().
+			Roles(r.Namespace).
+			Update(ctx, &r, meta.UpdateOptions{})
+		if err != nil {
+			return err
+		}
+	}
+
+	rb := rbac.RoleBinding{
+		ObjectMeta: meta.ObjectMeta{
+			Namespace: types.WalrusSystemNamespace,
+			Name:      types.DeployerServiceAccountName,
+		},
+		Subjects: []rbac.Subject{
+			{
+				Kind: rbac.ServiceAccountKind,
+				Name: types.DeployerServiceAccountName,
+			},
+		},
+		RoleRef: rbac.RoleRef{
+			APIGroup: rbac.GroupName,
+			Kind:     "Role",
+			Name:     types.DeployerServiceAccountName,
+		},
+	}
+
+	rb_, err := cli.RbacV1().
+		RoleBindings(rb.Namespace).
+		Get(ctx, rb.Name, meta.GetOptions{
+			ResourceVersion: "0",
+		})
+	if err != nil && !kerrors.IsNotFound(err) {
+		return err
+	}
+
+	switch {
+	case rb_ == nil || rb_.Name == "" || rb_.DeletionTimestamp != nil:
+		// Create.
+		_, err = cli.RbacV1().
+			RoleBindings(rb.Namespace).
+			Create(ctx, &rb, meta.CreateOptions{})
+		if err != nil {
+			return err
+		}
+	case !reflect.DeepEqual(rb.RoleRef, rb_.RoleRef):
+		// Delete.
+		err = cli.RbacV1().
+			RoleBindings(rb.Namespace).
+			Delete(ctx, rb.Name, meta.DeleteOptions{
+				PropagationPolicy: point(meta.DeletePropagationForeground),
```

Comment:
Can we get rid of the point func in this way: https://github.com/kubernetes/kubernetes/blob/35199e42a41f83520e640fbfa9f409516faa7501/cmd/kubeadm/app/phases/upgrade/health.go#L200

Reply:
updated